department-of-veterans-affairs/caseflow
# [Restored] TYLERB/APPEALS-9101 Make All Pre-Docket Modals' Submission Buttons Use 'usa-button' CSS Class

## Resolves [APPEALS-9101](https://vajira.max.gov/browse/APPEALS-9101) | [Original PR](https://github.com/department-of-veterans-affairs/caseflow/pull/17461)

### Description
Currently there are some modals used in pre-docket workflows using the secondary-default button styling, and others using the primary-default button styling. We would like all modals to be uniform and use the primary-default button styling. Currently, this change should only affect the pre-docket modals, but would later be expanded to be all of caseflow.

### Acceptance Criteria
- [ ] All buttons should have a hover effect
- [ ] Only pre-docket modals should be changed?
- [ ] All Submission Buttons on all pre-docket modals should be assigned the 'usa-button' class
  - [ ] CAMO Modals
    - [ ] Send to Board Intake (VHA CAMO)
    - [ ] Assign to Program Office (VHA CAMO)
  - [ ] VHA Program Office Modals
    - [ ] Ready for review (VHA PO)
    - [ ] Assign to VISN
    - [ ] Return to CAMO team (VHA PO)
    - [ ] Mark task in progress (VHA PO)
    - [ ] Begin Hold (VHA PO)
    - [ ] End Hold (VHA PO)
  - [ ] EMO Modals 
    - [ ] Ready for review (EMO)
    - [ ] Assign to RPO (EMO)  
    - [ ] Return to Board Intake (EMO)
  - [ ] Education RPO Modals 
    - [ ] Ready for review (Education RPO)
    - [ ] Return to Executive Management Office (EDU RPO)
    - [ ] Mark task in progress (EDU RPO)
  - [ ] BVA Intake Modals
    - [ ] Docket Appeal
    - [ ] Return appeal to VHA CAMO (BVA Intake)
    - [ ] Return appeal to Education Service (BVA Intake)

### Testing Plan
1. Setup some appeals and tasks to be used for testing
```ruby
5.times do
  appeal = FactoryBot.create(:appeal, :with_pre_docket_task)
  parent_task = appeal.tasks.last
  emo_document_search_task=EducationDocumentSearchTask.create!( parent: parent_task, appeal: appeal, assigned_at: Time.zone.now, assigned_to: EducationEmo.singleton )
end

5.times do
  appeal = FactoryBot.create(:appeal, :with_pre_docket_task)
  parent_task = appeal.tasks.last
  vha_document_search_task=VhaDocumentSearchTask.create!( parent: parent_task, appeal: appeal, assigned_at: Time.zone.now, assigned_to: VhaCamo.singleton )
end
```

2. Swap to a CAMO user.   There is one with the cssid of CAMOUSER. Navigate to the Camo organization queue.

3. Select one of the created appeals and go to the details page.

4. Select the Send to Board Intake task action and verify the button is the correct styling

5. Send one of the appeals back to intake for later testing

6. Select the Assign to Program Office task action and verify the button is the correct styling

7. Send a task/appeal to a program office of your choosing.

8. Switch to a program office user VHAPOUSER is the css id and navigate to the program office queue that you sent the appeal to

9. Select that appeal and go to the details page

10. Select the Ready for Review task action and verify the button is the correct styling

11. Select the Assign to VISN task action and verify the button is the correct styling

12. Select the Return to Camo team task action and verify the button is the correct styling

13. Select the Mark task in progress task action and verify the button is the correct styling

14. Select the Put task on hold task action and verify the button is the correct styling

15. Place the task on hold

16. Select the End hold early task action and verify the button is the correct styling
 
17. Switch to an emo user the css id is EMOUSER
18. Select an appeal and go to the details page
19. Select the Assign to regional processing office task action and verify the button is the correct styling
20. Send an appeal to an office of your choosing for po testing and select a new appeal
21. Select the Return to board intake task action and verify the button is the correct styling
22. Send a task back to intake for later testing and select a new appeal
23. Select the Ready for review task action and verify the button is the correct styling
24. Swap to an edu rpo user the css id is EDURPOUSER 
25. Go to the program office queue you sent the earlier appeal to
26. Select the appeal and go to the case details page
27. Select the Return to Executive Management Office task action and verify the button is the correct styling
28. Select the Ready for Review task action and verify the the button is the correct styling
29. Select the mark task in progress task action and verify the button is the correct styling
30. Swap to a BVA intake user css id is BVAISHAW
31. Navigate to the queue, go to the review tab and select the appeal returned from Camo
32. Select the Docket Appeal task action and verify the button is the correct styling
33. Select the Return appeal to VHA task action and verify the button is the correct styling
34. Navigate back to the queue, go to the review tab, and select the appeal returned from Emo
35. Select the Return appeal to Education Service task action and verify the button is the correct styling
